dts/bindings/flash_controller/infineon,xmc4xxx-flash-controller.yaml

```diff
@@ -0,0 +1,25 @@
+description: XMC4XXX flash controller
+
+compatible: "infineon,xmc4xxx-flash-controller"
+
+include: flash-controller.yaml
```

Comment:
Like the DTS, the binding seems to be mixing up the flash memory and the flash controller.

- `erase-value` looks like it should be a property of the flash memory, not the controller, but I see we are making the same mistake in zephyr,sim-flash.yaml, so it's OK to leave this as-is
- however, the flash page layout is definitely an attribute of the flash memory itself, not the controller, right? I think this should be handled in the flash then.

What I recommend is doing this:

- creating a new file, `dts/bindings/mtd/infineon,xmc4xxx-nv-flash.yaml`, which includes `soc-nv-flash.yaml` but uses its own compatible value
- moving the `child-binding` there as a new custom property for your SoC flash binding

Pinging @nvlsianpu as the flash maintainer to confirm this idea is OK.

Reply:
> erase-value looks like it should be a property of the flash memory, not the controller, but I see we are making the same mistake in zephyr,sim-flash.yaml, so it's OK to leave this as-is

Looks like this is only used by the zephyr,sim-flash.yaml. Reason is that we want to have it configurable. For real drivers we embedded this value in its flash driver implementations. And yes - it should be a flash property, although I never seen flashcontroller which control memories with different eras-value.
Is it really configurable for this driver. If not I would remove it.

Reply:
What @mbolivar-nordic suggested sounds ok to me.